src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -81,27 +81,18 @@ def create_client(self) -> AuthenticatedClient:
         logger.info("Creating client to %s", uri)
         return AuthenticatedClient(uri, self.configuration)
 
-    def _build_client_with_config(
-        self, client_config: ClientConfig
-    ) -> AuthenticatedClient:
-        """Sets the client_config on the SaasConnector, and also sets it on the created AuthenticatedClient"""
-        self.client_config = client_config
-        client: AuthenticatedClient = self.create_client()
-        client.client_config = client_config
-        return client
-
-    def create_client_from_request(
+    def create_client_with_request(
         self, saas_request: SaaSRequest
     ) -> AuthenticatedClient:
         """
-        Permits authentication to be overridden at the request-level.
-        Use authentication on the request if specified, otherwise, just use
-        the authentication configured for the overall SaaS connector.
+        Permits configuration at the request-level for the connector and client
+        Overrides client_config in the current connector if set in request
         """
         if saas_request.client_config:
-            return self._build_client_with_config(saas_request.client_config)
-
-        return self._build_client_with_config(self.saas_config.client_config)  # type: ignore
+            self.client_config = saas_request.client_config
+        client: AuthenticatedClient = self.create_client()
+        client.saas_request = saas_request
+        return client
```

Comment:
I also wanted to take a stab and potentially simplifying this logic. Rather than having extra logic here to pass in the correct config, I think we can just pass the `SaaSRequest` and let `AuthenticatedClient` determine which config to use. This way we can also avoid passing in `ignore_errors` and `rate_limits` in the `send` call. 

It's a little weird that we set `self.client_config` here but it doesn't look like there is an easy way around that rn.

Reply:
The issue is that `create_client` is that it is inherited and doesn't take in any parameters and within the lifecycle of a `SaaSConnector` there is many `SaaSRequest` instances. So I think this leaves us with a scenario where we have to change the state of the class if we want to be able to use the `create_client` method. 

I actually ended up undoing these changes because I found some usages of AuthenticatedClient that I couldn't reconcile with this refactor. For instance, in OAuth2AuthenticationStrategyBase we use the client but it doesn't make send to use a saas_request there. This code could use some refactor but I don't have a clean idea atm.